src/main/java/uk/gov/hmcts/darts/common/repository/ObjectRetrievalQueueRepository.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.darts.common.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectRetrievalQueueEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+
+import java.util.Optional;
+
+public interface ObjectRetrievalQueueRepository extends JpaRepository<ObjectRetrievalQueueEntity, Integer> {
+
+    @Query(
+        """
+            SELECT orq FROM ObjectRetrievalQueueEntity orq
+            WHERE orq.parentObjectId = :parentObjectId
+            AND orq.contentObjectId = :contentObjectId
+            AND orq.clipId = :clipId
+            AND (orq.media = :media
+            or orq.transcriptionDocument = :transcriptionDocument)
+        """
+    )
+    Optional<ObjectRetrievalQueueEntity> findMatchingObjectRetrievalQueuedItems(MediaEntity media,
```

Comment:
is this expected to return more than 1 - just curious re: naming of "QueuedItems"

Reply:
Good point, it will only return one item. Have renamed to findMatchingObjectRetrievalQueueItem